src/main/java/org/apache/mesos/offer/DynamicPortRequirement.java

```diff
@@ -0,0 +1,95 @@
+package org.apache.mesos.offer;
+
+import org.apache.mesos.Protos;
+
+/**
+ * This class allows the expression of a desire for a random port designated by a name which will be injected into the
+ * environment.
+ */
+public class DynamicPortRequirement extends ResourceRequirement {
+    private static final String DYNAMIC_PORT_KEY = "dynamic_port_key";
```

Comment:
Can just go with DYNAMIC_PORT_KEY='dynamic_port' imo, also affects test values in OfferEvaluatorTest. The protobuf will display this as { key: dynamic_port, value: foo }